Core/Core/Inbox/AttachmentPicker/ViewModel/AttachmentPickerViewModel.swift

```diff
@@ -0,0 +1,185 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import Combine
+import CombineExt
+
+class AttachmentPickerViewModel: ObservableObject {
+
+    // MARK: Inputs / Outputs
+
+    @Published public var isAttachmentSelectorVisible: Bool = false
+    @Published public var isFilePickerVisible: Bool = false
+    @Published public var isImagePickerVisible: Bool = false
+    @Published public var isTakePhotoVisible: Bool = false
+    @Published public var isAudioRecordVisible: Bool = false
+    @Published public var fileList: [File] = []
+    public let title = NSLocalizedString("Attachments", comment: "")
+    public let cancelButtonDidTap = PassthroughRelay<WeakViewController>()
+    public let doneButtonDidTap = PassthroughRelay<WeakViewController>()
+    public let uploadButtonDidTap = PassthroughRelay<WeakViewController>()
+    public let retryButtonDidTap = PassthroughRelay<WeakViewController>()
+    public let addAttachmentButtonDidTap = PassthroughRelay<WeakViewController>()
+    public let router: Router
+
+    // MARK: Outputs
+
+    public var isError: Bool {
+        fileList.contains(where: { file in file.uploadError != nil })
+    }
+
+    public var isUploading: Bool {
+        fileList.contains(where: { file in file.isUploading })
+    }
+
+    public var isAllUploaded: Bool {
+        !fileList.contains(where: { file in !file.isUploaded })
+    }
```

Comment:
We could write an extension for file sequences and then we don't have this logic in the view model. 

```
extension Sequence where Element == File {
    public var isLoading: Bool { ... }
}
```